Linux: PATH makes it from /etc/profile.d but not LD_LIBRARY_PATH

## Having issue with YAML?
N/A

## Having issue with Tasks?
N/A

## Having issue with software on Hosted Agent?
N/A

## Having generic issue with Azure-Pipelines/VSTS/TFS?
N/A

## Have you tried trouble shooting?
N/A

## Agent Version and Platform
 2.141.2, Linux x86_64 (RHEL7)

## VSTS Type and Version
VisualStudio.com, account usfhii

## What's not working?
Our agents have git 2.9.3 installed via RH SCL, which requires `$PATH` and `$LD_LIBRARY_PATH` to be set to function. I have an entry in `/etc/profile.d/*` that provides both as follows. `$PATH` seems to make it into the agent's runtime, but `$LD_LIBRARY_PATH` does not, causing SCL software (eg, git) not to work when executed by the agent. Presumably `$PERL5LIB` wouldn't make it, either.

This causes the "Get sources" phase to fall over, before any agent jobs (thus tasks) can be run.

```
# enable software collections for git 2.9.x
if [ -x /usr/bin/scl_source ]; then
  if [ -r /etc/scl/prefixes/rh-git29 ]; then
    source scl_source enable rh-git29
  fi
fi
```

This results in the following:
```
[redacted@redacted ~]$ env | grep /opt/rh
MANPATH=/opt/rh/rh-git29/root/usr/share/man:
PERL5LIB=/opt/rh/rh-git29/root/usr/share/perl5/vendor_perl
LD_LIBRARY_PATH=/opt/rh/httpd24/root/usr/lib64
PATH=redacted:/opt/rh/rh-git29/root/usr/bin:redacted
```

## Agent and Worker's Diagnostic Logs
Log output from VisualStudio.com:

```
2018-12-03T20:19:29.3701420Z ##[section]Starting: Checkout
2018-12-03T20:19:29.3945235Z ==============================================================================
2018-12-03T20:19:29.3945374Z Task         : Get sources
2018-12-03T20:19:29.3945529Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2018-12-03T20:19:29.3945570Z Version      : 1.0.0
2018-12-03T20:19:29.3945603Z Author       : Microsoft
2018-12-03T20:19:29.3945653Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
2018-12-03T20:19:29.3945711Z ==============================================================================
2018-12-03T20:19:30.0257864Z Syncing repository: core-ssocas (Git)
2018-12-03T20:19:30.0291408Z ##[command]git version
2018-12-03T20:19:30.0292571Z git version 2.9.3
2018-12-03T20:19:30.0293038Z ##[command]git lfs version
2018-12-03T20:19:30.0797657Z git-lfs/2.5.2 (GitHub; linux amd64; go 1.10.3; git 8e3c5c93)
2018-12-03T20:19:30.1126337Z ##[command]git init "/opt/vsts/agent/_work/1/s"
2018-12-03T20:19:30.1382384Z Initialized empty Git repository in /opt/vsts/agent/_work/1/s/.git/
2018-12-03T20:19:30.1427570Z ##[command]git remote add origin https://REDACTED
2018-12-03T20:19:30.1485041Z ##[command]git config gc.auto 0
2018-12-03T20:19:30.1539333Z ##[command]git config --get-all http.https://REDACTED.extraheader
2018-12-03T20:19:30.1578192Z ##[command]git config --get-all http.proxy
2018-12-03T20:19:30.1676404Z ##[command]git -c http.extraheader="AUTHORIZATION: bearer ***" fetch --tags --prune --progress --no-recurse-submodules origin
2018-12-03T20:19:30.2051316Z git-remote-https: error while loading shared libraries: libcurl-httpd24.so.4: cannot open shared object file: No such file or directory
2018-12-03T20:19:30.2124481Z ##[warning]Git fetch failed with exit code 128, back off 5.003 seconds before retry.
2018-12-03T20:19:35.2108397Z ##[command]git -c http.extraheader="AUTHORIZATION: bearer ***" fetch --tags --prune --progress --no-recurse-submodules origin
2018-12-03T20:19:35.2134758Z git-remote-https: error while loading shared libraries: libcurl-httpd24.so.4: cannot open shared object file: No such file or directory
2018-12-03T20:19:35.2136062Z ##[warning]Git fetch failed with exit code 128, back off 6.622 seconds before retry.
2018-12-03T20:19:41.8368462Z ##[command]git -c http.extraheader="AUTHORIZATION: bearer ***" fetch --tags --prune --progress --no-recurse-submodules origin
2018-12-03T20:19:41.8393344Z git-remote-https: error while loading shared libraries: libcurl-httpd24.so.4: cannot open shared object file: No such file or directory
2018-12-03T20:19:41.8566225Z ##[error]System.InvalidOperationException: Git fetch failed with exit code: 128
   at Agent.Plugins.Repository.GitSourceProvider.GetSourceAsync(AgentTaskPluginExecutionContext executionContext, RepositoryResource repository, CancellationToken cancellationToken)
   at Agent.Plugins.Repository.CheckoutTask.RunAsync(AgentTaskPluginExecutionContext executionContext, CancellationToken token)
   at Agent.PluginHost.Program.Main(String[] args)
2018-12-03T20:19:41.8663179Z ##[section]Finishing: Checkout
```

--- a/src/Misc/layoutroot/env.sh
+++ b/src/Misc/layoutroot/env.sh
@@ -12,6 +12,8 @@ varCheckList=(
     'VSTS_HTTP_PROXY'
     'VSTS_HTTP_PROXY_USERNAME'
     'VSTS_HTTP_PROXY_PASSWORD'
+    'LD_LIBRARY_PATH'
+    'PERL5LIB'
     )
 
 envContents=""

